spec/bundler/ui/shell_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+RSpec.describe Bundler::UI::Shell do
+  subject { described_class.new }
+
+  before { subject.level = "debug" }
+
+  describe "#info" do
+    before { subject.level = "info" }
+    it "prints to stdout" do
+      expect { subject.info("info") }.to output("info\n").to_stdout
+    end
+  end
+
+  describe "#confirm" do
+    before { subject.level = "confirm" }
+    it "prints to stdout" do
+      expect { subject.confirm("confirm") }.to output("confirm\n").to_stdout
+    end
+  end
+
+  describe "#warn" do
+    before { subject.level = "warn" }
+    it "prints to stdout" do
+      expect { subject.warn("warning") }.to output("warning\n").to_stdout
+    end
+  end
+
+  describe "#debug" do
+    it "prints to stdout" do
+      expect { subject.debug("debug") }.to output("debug\n").to_stdout
+    end
+  end
+
+  describe "#error" do
+    before { subject.level = "error" }
+    it "prints to stdout" do
+      expect { subject.error("error!!!") }.to output("error!!!\n").to_stdout
+    end
+
+    context "when stderr flag is enabled" do
+      before { bundle "config error_on_stderr true" }
+      it "prints to stderr" do
+        expect { subject.error("error!!!") }.to output("\e[31merror!!!\e[0m").to_stderr
```

Comment:
I'm not entirely sure what's happening with the ANSI encoding, if i print to stdout out instead the tests pass. 🤷‍♀️ 

```
diff --git a/lib/bundler/ui/shell.rb b/lib/bundler/ui/shell.rb
index 81743ede4..2ee37d4e6 100644
--- a/lib/bundler/ui/shell.rb
+++ b/lib/bundler/ui/shell.rb
@@ -111,8 +111,8 @@ module Bundler
         buffer = @shell.send(:prepare_message, message, *color)
         buffer << "\n" if newline && !message.to_s.end_with?("\n")
 
-        @shell.send(:stderr).print(buffer)
-        @shell.send(:stderr).flush
+        @shell.send(:stdout).print(buffer)
+        @shell.send(:stdout).flush
       end
 
       def strip_leading_spaces(text)
diff --git a/spec/bundler/ui/shell_spec.rb b/spec/bundler/ui/shell_spec.rb
index 2db2d2864..ce76d67aa 100644
--- a/spec/bundler/ui/shell_spec.rb
+++ b/spec/bundler/ui/shell_spec.rb
@@ -42,7 +42,7 @@ RSpec.describe Bundler::UI::Shell do
     context "when stderr flag is enabled" do
       before { bundle "config error_on_stderr true" }
       it "prints to stderr" do
-        expect { subject.error("error!!!") }.to output("\e[31merror!!!\e[0m\n").to_stderr
+        expect { subject.error("error!!!") }.to output("error!!!\n").to_stdout
       end
     end
   end
```

```
› rspec spec/bundler/ui/shell_spec.rb -f progress
Run options: exclude {:rubygems_master=>true, :git=>"=< 2.13.0", :rubygems=>"=< 2.6.8", :ruby=>"=< 2.4.0", :realworld=>true, :sudo=>true}
......

Retried examples: 0


Finished in 0.85979 seconds (files took 0.27336 seconds to load)
6 examples, 0 failures

```

Any ideas?